Aardvark/ARKEmailBugReporter.m

```diff
@@ -37,6 +37,82 @@ @interface ARKInvisibleView : UIView
 @end
 
 
+@interface CALayer (HierarchyDescription)
+
+- (NSString *)_recursiveLayerHierarchyDescriptionWithIndentationLevel:(int)indentationLevel;
+
+@end
+
+
+@implementation CALayer (HierarchyDescription)
+
+- (NSString *)_recursiveLayerHierarchyDescriptionWithIndentationLevel:(int)indentationLevel;
+{
+    NSMutableString *const mutableDescription = [NSMutableString new];
+    for (int i = 0 ; i < indentationLevel ; i++) {
+        [mutableDescription appendString:@"   | "];
+    }
+    
+    [mutableDescription appendFormat:@"%@\n", [self description]];
+    
+    for (CALayer *sublayer in self.sublayers) {
+        [mutableDescription appendString:[sublayer _recursiveLayerHierarchyDescriptionWithIndentationLevel:(indentationLevel + 1)]];
+    }
+    
+    return mutableDescription;
+}
+
+@end
+
+
+@interface UIView (HierarchyDescription)
+
+/// Re-implementation of `-[UIView recursiveDescription]` to avoid calling private API.
+- (NSString *)recursiveViewHierarchyDescription;
+
+@end
+
+
+@implementation UIView (HierarchyDescription)
+
+- (NSString *)recursiveViewHierarchyDescription;
+{
+    return [self _recursiveViewHierarchyDescriptionWithIndentationLevel:0];
+}
+
+#pragma mark - Private Methods
+
+- (NSString *)_recursiveViewHierarchyDescriptionWithIndentationLevel:(int)indentationLevel;
+{
+    NSMutableString *const mutableDescription = [NSMutableString new];
+    for (int i = 0 ; i < indentationLevel ; i++) {
+        [mutableDescription appendString:@"   | "];
+    }
+    
+    [mutableDescription appendFormat:@"%@\n", [self description]];
```

Comment:
curious: do you need to call `description` here?

Reply:
Indeed you do not. :-)

Reply:
Originally had it calling `debugDescription`, but changed it.  Will remove.